Content.Shared/SprayPainter/SprayPainterEvents.cs

```diff
@@ -71,3 +91,19 @@ public SprayPainterPipeDoAfterEvent(Color color)
 
     public override DoAfterEvent Clone() => this;
 }
+
+/// <summary>
+/// An action raised on an item when it was spray painted.
+/// </summary>
+[ByRefEvent]
+public partial record struct EntityPaintedEvent(EntityUid? user, EntityUid tool, EntProtoId prototype, ProtoId<PaintableGroupPrototype> group)
+{
+    /// <summary>The entity painting this item.</summary>
+    public EntityUid? User = user;
+    /// <summary>The entity used to paint this item.</summary>
+    public EntityUid Tool = tool;
+    /// <summary>The prototype used to generate the new painted appearance.</summary>
+    public EntProtoId Prototype = prototype;
+    /// <summary>The group of item being painted (e.g. airlocks with glass, canisters).</summary>
+    public ProtoId<PaintableGroupPrototype> Group = group;
+}
```

Comment:
```suggestion
/// <summary>
/// An action raised on an item when it was spray painted.
/// </summary>
/// <param name="User">The entity painting this item.</param>
/// <param name="Tool">The entity used to paint this item.</param>
/// <param name="Prototype">The prototype used to generate the new painted appearance.</param>
/// <param name="Group">The group of item being painted (e.g. airlocks with glass, canisters).</param>
[ByRefEvent]
public partial record struct EntityPaintedEvent(
    EntityUid? User,
    EntityUid Tool,
    EntProtoId Prototype,
    ProtoId<PaintableGroupPrototype> Group);
```

Record structs don't need member fields.

Reply:
Fair dues.  Changes applied.